common-npm-packages/packaging-common/npm/npmregistry.ts

```diff
@@ -59,31 +60,31 @@ export class NpmRegistry implements INpmRegistry {
                 username = endpointAuth.parameters['username'];
                 password = endpointAuth.parameters['password'];
                 email = username; // npm needs an email to be set in order to publish, this is ignored on npmjs
-                password64 = (new Buffer(password).toString('base64'));
-                tl.setSecret(password64);
+                password64 = Buffer.from(password).toString('base64');
 
                 auth = nerfed + ':username=' + username + lineEnd;
                 auth += nerfed + ':_password=' + password64 + lineEnd;
                 auth += nerfed + ':email=' + email + lineEnd;
                 break;
             case 'Token':
                 const apitoken = endpointAuth.parameters['apitoken'];
+                tl.setSecret(apitoken);
                 if (!isVstsTokenAuth){
                     // Use Bearer auth as it was intended.
                     auth = nerfed + ':_authToken=' + apitoken + lineEnd;
                 } else {
                     // Azure DevOps does not support PATs+Bearer only JWTs+Bearer
                     email = 'VssEmail';
                     username = 'VssToken';
-                    password64 = (new Buffer(apitoken).toString('base64'));
```

Comment:
should this password field be entirely removed ?

Reply:
oops I accidently deleted it, added back,